### PR TITLE
Pr/42

### PR DESCRIPTION
**Pull request by ShaneMPutnam**
This PR includes the workflow developed by @Mark-S-Bartlett for calculating a stratified sample of runoff events given rainfall and maximum potential retention distributions. In its current state, the workflow produces reasonable results and can be used for performing necessary calculations. Still, issues will be added so that enhancements can be made once the code is pulled into the master.

### CODE REVIEW

File: core/hydromet_stratified.py [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+from scipy import interpolate, stats, special
+from matplotlib import pyplot as plt
+from cycler import cycler
+
+
+#----------------------------------------------------------------------------------------------------------------------#
+# Functions called by EventTable_Stratified.ipynb.
+#----------------------------------------------------------------------------------------------------------------------#
+
+def Q_SCS(R: np.array, CN: float, mu: float) -> float:
```
Comment (slawler) [inline core/hydromet_stratified.py:15]: Let's make the R variable more meaningful, e.g. 'rainfall_total' and then add a comment with the units and any additional info. The description could be then copied to better comment the other function docstrings.

File: core/hydromet_stratified.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+
+def Qzero_integrand(S: float, mu: float, alpha: float, beta: float, S_limit: float, GEV_parameters: np.ndarray,
+                    PMP: float, error_PQ: float) -> float:
+    """Defines the integrand for calculating the probability of zero runoff.
+    """
+    return (CDF_GEV(S*mu, GEV_parameters, PMP)-CDF_GEV(0, GEV_parameters, PMP))\
+           *(1.0/S_limit)*stats.beta(alpha, beta).pdf(S/S_limit)/(1-error_PQ)
+
+
+def P_Qzero(mu: float, alpha: float, beta: float, S_limit: float, GEV_parameters: np.ndarray, PMP: float,
```
Comment (slawler) [inline core/hydromet_stratified.py:138]: General comment: where there is a parameter like S_Limit, please add a descriptor and an example value or range of acceptable values.

File: core/hydromet_stratified.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
+import pathlib as pl
+import numpy as np
+import pandas as pd
+from scipy.integrate import quad
+from scipy.optimize import minimize
+from scipy import interpolate, stats, special
+from matplotlib import pyplot as plt
+from cycler import cycler
+
```
Comment (slawler) [inline core/hydromet_stratified.py:9]: You might consider putting some info here describing the scope of this file.

File: core/hydromet_stratified.py [hunk 4]
```diff
@@ -0,0 +1,560 @@
...
+    """
+    S = 1000.0/CN-10.0
+    return (R-mu*S)**2/(R-mu*S+S)
+
+
+def Norm_Constant_GEV(x: np.ndarray, PMP: float) -> float:
+    """Constant for distribution truncation at the PMP value.
+    """ 
+    return 1.0/stats.genextreme.cdf(PMP, x[2], x[0], x[1])
+
```
[the remark this answers is not in the document]
Comment (ShaneMPutnam) [reply core/hydromet_stratified.py:26]: Added as issue #52.

File: core/hydromet_stratified.py (hunk 1 above)
Comment (ShaneMPutnam) [reply core/hydromet_stratified.py:15]: Added to issue #47.

File: core/hydromet_stratified.py (hunk 3 above)
Comment (ShaneMPutnam) [reply core/hydromet_stratified.py:9]: Added as issue #53.

**Pull request merged**